src/main/webapp/script.js

```diff
@@ -65,69 +65,78 @@ async function onSubmitUploadImageForm() {
+    console.warn(error);
+    return new Response(JSON.stringify({
+      code: error.response.status,
+      message: `Failed to fetch ${imageUploadUrl}`,
+    }));
+  });
 
   if (!response.ok) {
     return Promise.reject(response);
   }
 
-  // The request returns a JSON array with the shopping query used to search on Google Shopping and
-  // the data about each product from the Google Shopping results page.
-  const data = await response.json();
-  const shoppingQuery = data[0]; 
-  const products = data[1]; 
+  // The request returns a JSON object with the all the shopping queries used to search on Google Shopping and
+  // the data about each product for each query from the Google Shopping results page.
+  const shoppingResults = await response.json();
 
-  // Empty the prompt container and add the {@code products} content into the page.
+  // Empty the prompt container and add the query and its resultant content into the page.
   $('#search-loading-prompt').empty();
   $('#loading-gif-prompt').addClass('hidden');
 
-  // Show the user the shopping query built to search on Google Shopping.
   $('#shopping-query-display-container').removeClass('hidden');
-  $('#shopping-query').text(shoppingQuery);
+  // Add a new line before displaying the queries.
+  $('#shopping-query').append('<br />');
 
-  appendShoppingResults(products);
+  appendShoppingResults(shoppingResults);
 }
 
 /**
- * Integrates product results from Google Shopping into the web page. 
+ * Integrates product results from Google Shopping and their respective queries into the web page.
  */
-async function appendShoppingResults(products) {
-  // Integrate the products into the web page.
-  products.forEach(product => {
-    // Create an HTML node for the item container.
-    let $productContainer = $('<div>', {class: 'col-md-4'});
-
-    // Get the HTML content for the container.
-    const productElementHTML = getProductElementHTML(product.title,
-                                                     product.imageLink,
-                                                     product.priceAndSeller,
-                                                     product.link,
-                                                     product.shippingPrice);
-
-    // Load the content using jQuery's append.
-    $productContainer.append(productElementHTML);
-
-    // Add the container to the results page, into the corresponding product wrapper.
-    $('#shopping-results-wrapper').append($productContainer);
+async function appendShoppingResults(shoppingResults) {
+  shoppingResults.forEach((result) => {
+    // Show the user the shopping query built to search on Google Shopping.
+    $('#shopping-query').append(result['query'] + '<br />');
+    const $query = `<div class='text-center query'>${result['query']}</div>`;
+    $('#shopping-results-wrapper').append($query);
+
+    const $listProductsContainer = $('<div>', {class: 'row'});
+
+    const $products = result['products'];
```

Comment:
Don't use `$` in your variable names.

Reply:
So, should I not use `$` for any variable in the whole .js file?

Reply:
Google's style guide says: "Identifiers use only ASCII letters and digits, and, in a small number of cases noted below, underscores and very rarely (when required by frameworks like Angular) dollar signs."
https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers

I know there are other conventions out there, but it's good to stick to one style guide. If you're following another style guide and you use the dollar signs consistently that's also fine.